spec.html

```diff
@@ -32497,173 +32497,107 @@ <h1>String.prototype.replace ( _searchValue_, _replaceValue_ )</h1>
           1. Let _searchLength_ be the length of _searchString_.
           1. Let _position_ be ! StringIndexOf(_string_, _searchString_, 0).
           1. If _position_ is -1, return _string_.
-          1. Let _preserved_ be the substring of _string_ from 0 to _position_.
+          1. Let _preceding_ be the substring of _string_ from 0 to _position_.
+          1. Let _following_ be the substring of _string_ from _position_ + _searchLength_.
           1. If _functionalReplace_ is *true*, then
             1. Let _replacement_ be ? ToString(? Call(_replaceValue_, *undefined*, &laquo; _searchString_, 𝔽(_position_), _string_ &raquo;)).
           1. Else,
             1. Assert: Type(_replaceValue_) is String.
             1. Let _captures_ be a new empty List.
             1. Let _replacement_ be ! GetSubstitution(_searchString_, _string_, _position_, _captures_, *undefined*, _replaceValue_).
-          1. Return the string-concatenation of _preserved_, _replacement_, and the substring of _string_ from _position_ + _searchLength_.
+          1. Return the string-concatenation of _preceding_, _replacement_, and _following_.
         </emu-alg>
         <emu-note>
           <p>The `replace` function is intentionally generic; it does not require that its *this* value be a String object. Therefore, it can be transferred to other kinds of objects for use as a method.</p>
         </emu-note>
 
-        <emu-clause id="sec-getsubstitution" type="abstract operation">
+        <emu-clause id="sec-getsubstitution" type="abstract operation" oldids="table-replacement-text-symbol-substitutions,table-45">
           <h1>
             GetSubstitution (
               _matched_: unknown,
               _str_: unknown,
               _position_: a non-negative integer,
               _captures_: unknown,
               _namedCaptures_: unknown,
-              _replacement_: unknown,
+              _replacer_: unknown,
```

Comment:
Hmm, "replacer" kind of implies a function to me (especially since you can pass a function in the place where you would provide this string).

Maybe `replacementTemplate`? A bit overloaded with template literals, I realize, but that's what it is: a template for deriving the replacement string.

Reply:
I wanted to use "template", but didn't because of the clash with Template Literals. If you think it's not a problem, I'll make that change.

Reply:
+1 on not "replacer". i'm not sure "template" makes much sense either - it's not really a template either (into which things are interpolated) - it's a replacement string, which might be inserted multiple times (and the return value is the replaced string, for which "replacement" also kind of works).

I don't have a suggestion tho.

Reply:
It is a template into which things are interpolated. Those things are determined by the `$` escapes described in the if/else below.

Reply:
Ah, true, hmm

Reply:
I changed `_replacer_` to `_replacementTemplate_` as suggested.
And `_replacerRemainder_` to `_templateRemainder_`.

I also reverted my `_result_` -> `_replacement_` renaming, as I think `_result_` works better here.